packages/bpk-component-graphic-promo/package.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name": "bpk-component-graphic-promo",
+  "version": "1.0.0",
+  "description": "Backpack graphic-promo component.",
+  "license": "Apache-2.0",
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:Skyscanner/backpack.git"
+  },
+  "author": "Backpack Design System <backpack@skyscanner.net>",
+  "main": "index.js",
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/"
+  },
+  "dependencies": {
+    "bpk-component-button": "^5.1.0",
+    "bpk-component-card": "^4.1.1",
+    "bpk-component-text": "^6.1.1",
+    "bpk-mixins": "^26.1.0",
+    "bpk-react-utils": "^4.1.0",
+    "prop-types": "^15.5.8"
+  },
+  "peerDependencies": {
+    "react": "^16.3.0 || ^17.0.0"
```

Comment:
Can be just `^17.0.0`:
- Only intended to be used in one place in the short term, and that place used 17
- Internal standard has changed to 17, so is fair for any new consumers to be required to update in the medium term

Reply:
I guess if we aren't using anything specific out of 17 this can be fine as per the generated template as the full 17 upgrade hasn't finished yet and it means that we can still show support without warnings for those that may need to utilise this component before upgrading

Reply:
There is a line further down:

```
// FIXME: Use useCallback() here when React is updated.
```

Rather than having the FIXME, or removing it but using outdated syntax, I think we can just make `17` the minimum here and the fix of the fixme could include using the hook (if we decided it was the right solution) - no benefit to generate extra work or limit ourselves when we know the single consumer of this component is on 17. Backpack can act as a push here rather than a pull as it has to in existing components, since we know the exact single place this will be used. Very different from if writing a component intended to be used multiple places - this one may be _later_, but that _later_ is far enough away not to limit ourselves to syntax far older than the codebase this will be used in, IMO.

We can stick to 16.3, but to me it holds us in the past without a benefit - in the context of this component.

Reply:
>Backpack can act as a push here rather than being pulled as it has to in existing components, since we know the exact single place this will be used. Very different from if writing a component intended to be used multiple places - this one may be later, but that later is far enough away not to limit ourselves to syntax far older than the codebase this will be used in, IMO.

This is not true as by introducing this component now I know there are other vertical teams that will start using this soon/now which aren't on 17 so we need to keep this backwards compatibility until the React 17 upgrade is complete. 

We can add updating this component to our React 17 upgrade list for when the time comes which is fine for us but for now we don't want to force people to duplicate the work/not use Backpack as we are supposed to be the path of least resistance.

>If we stick on 16.3 we likely want to just remove the FIXME comment for cleanliness?

I would keep this for now so that it acts as a reminder in the team that we need/should update it in the future when we support this newer version/features as I guarantee this will be forgotten about otherwise

Reply:
👍

Reply:
I see why it would make sense to use React 17, since graphic promos will probably not be used by anyone other than Banana in the near future. That would also help fix the missing `useCallback` in the codebase. But since agreement was reached, I will leave it as it is for now. 🙂